### DISCUSSION TITLE
How to find needed endpoint?

### DISCUSSION THREAD

**Discussion by greensnaky** [oobabooga/text-generation-webui#4578] [Q&A]
Why is the number of endpoints so large (~250)? I cant find which is prompt and answer... I tried to use gradio_client and client.predict, but i dont know which fn_index i shoud use. moreover, in most places on huggingface, the number of endpoints does not exceed even 2, and there are parameters and returns everywhere.
![7SX2MhZUl6](https://github.com/oobabooga/text-generation-webui/assets/69247396/ece0abaa-b1c2-4849-b1b2-2672c86b891b)
help please

**Comment by simcop2387** [comment]
I don't think the gradio api like that is considered stable between updates so you probably want to look at using the built in OpenAI API since that's now the official way to interact with the server without the UI now.
